### PR TITLE
adds the ability to keybind click() triggers

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Adds the ability to keybind click options

## Why It's Good For The Game
Middle click is the big one here, lets trackpad users more easily point. I think having the ability to just get a examine/pull button is good

## Testing
Used them in game
realized rev is silly

## Changelog
:cl:
add: Click combinations can now be binded to a button
/:cl:

### CODE REVIEW

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -164,6 +164,9 @@
 
 	return
 
+/mob/proc/can_use_clickbinds()
+	return TRUE
```
Comment (Sirryan2002) [inline code/_onclick/click.dm:168]: document this

File: code/datums/keybindings/click_keybindings.dm [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	category = KB_CATEGORY_CLICK
+
+/datum/keybinding/clickbind/can_use(client/C, mob/M)
+	if(M.next_click > world.time)
+		return
+	M.changeNext_click(1)
+	return M.can_use_clickbinds() && ..()
+
+/datum/keybinding/clickbind/down(client/C)
+	. = ..()
```
Comment (Sirryan2002) [inline code/datums/keybindings/click_keybindings.dm:11]:
```suggestion
	..()
```
at no point is `.` returned in this proc

File: code/_onclick/click.dm [hunk 3]
```diff
@@ -164,6 +164,9 @@
 
 	return
```
Comment (GDNgit) [inline code/_onclick/click.dm:166]:
```suggestion

/// Can this mob use keybinded click actions? (Altclick, Ctrlclick, ect)
```

File: code/_onclick/click.dm (hunk 1 above)
Comment (GDNgit) [reply code/_onclick/click.dm:169]: Documented

**Review by Sirryan2002 — APPROVED** [review]
this looks fine? I don't know enough about click() stuff to say anything other than that

**Review by S34NW — APPROVED** [review]
This has worked well enough on live for it to go in

**Pull request merged**